lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+    ValueSourceError(#[from] ValueSourceError),
+    #[error(transparent)]
+    WorkspaceSnapshotError(#[from] WorkspaceSnapshotError),
+}
+impl AttributePrototypeDebugView {
+    #[instrument(level = "debug", skip_all)]
+    pub async fn new_from_attribute_value(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+    ) -> AttributePrototypeDebugViewResult<AttributePrototypeDebugView> {
+        let prototype_id = AttributeValue::prototype_id(ctx, attribute_value_id).await?;
+        let destination_component_id =
+            AttributeValue::component_id(ctx, attribute_value_id).await?;
+        let apa_ids = AttributePrototypeArgument::list_ids_for_prototype(ctx, prototype_id).await?;
+        let func_id = AttributePrototype::func_id(ctx, prototype_id).await?;
+        let all_attribute_values =
+            AttributePrototype::attribute_value_ids(ctx, prototype_id).await?;
+        let mut func_binding_args: HashMap<String, Vec<serde_json::Value>> = HashMap::new();
+        let mut arg_sources: HashMap<String, Option<String>> = HashMap::new();
+
+        let func_name = Func::get_by_id(ctx, func_id).await?.name;
+
+        let func_execution =
+            Some(FuncExecution::get_latest_execution_by_func_id(ctx, &func_id).await?);
+        for apa_id in apa_ids {
+            let apa = AttributePrototypeArgument::get_by_id(ctx, apa_id).await?;
+            let expected_source_component_id = apa
+                .targets()
+                .map(|targets| targets.source_component_id)
+                .unwrap_or(destination_component_id);
+
+            if apa.targets().map_or(true, |targets| {
+                targets.destination_component_id == destination_component_id
+            }) {
+                // If the "source" Component is marked for deletion, and we (the destination) are
+                // *NOT*, then we should ignore the argument as data should not flow from things
+                // that are marked for deletion to ones that are not.
+                let destination_component = Component::get_by_id(ctx, destination_component_id)
+                    .await
+                    .map_err(|e| AttributeValueError::Component(e.to_string()))?;
+
+                let source_component = Component::get_by_id(ctx, expected_source_component_id)
+                    .await
+                    .map_err(|e| AttributeValueError::Component(e.to_string()))?;
+
+                if source_component.to_delete() && !destination_component.to_delete() {
+                    continue;
+                }
+
+                let func_arg_id =
+                    AttributePrototypeArgument::func_argument_id_by_id(ctx, apa_id).await?;
+                let func_arg_name = ctx
+                    .workspace_snapshot()?
+                    .get_node_weight_by_id(func_arg_id)
+                    .await?
+                    .get_func_argument_node_weight()?
+                    .name()
+                    .to_owned();
```

Comment:
This should be behind a method in `FuncArgument` rather than using the snapshot directly.